src/rgw/driver/rados/rgw_service.cc

```diff
@@ -113,22 +114,22 @@ int RGWServices_Def::init(CephContext *cct,
 
   can_shutdown = true;
 
-  int r = finisher->start(y, dpp);
+  int r = finisher->start(y, dpp, cfgstore);
```

Comment:
this passes `cfgstore` to a bunch of classes that don't need it. it's probably much easier to pass `cfgstore` into the constructors of the classes that actually need it. that appears to be just these two?
```c++
  mdlog = std::make_unique<RGWSI_MDLog>(cct, run_sync);
  zone = std::make_unique<RGWSI_Zone>(cct);
```
that way you don't need to change `RGWServiceInstance` or any of its overrides